src/MagnumPlugins/WebPImporter/WebPImporter.cpp

```diff
@@ -0,0 +1,149 @@
+    /* Because here we're copying the data and using the _in to check if file
+       is opened, having them nullptr would mean openData() would fail without
+       any error message. It's not possible to do this check on the importer
+       side, because empty file is valid in some formats (OBJ or glTF). We also
+       can't do the full import here because then doImage2D() would need to
+       copy the imported data instead anyway (and the uncompressed size is much
+       larger). This way it'll also work nicely with a future openMemory(). */
+    if(data.isEmpty()) {
+        Error{} << "Trade::WebPImporter::openData(): the file is empty";
+        return;
+    }
+
+    /* Take over the existing array or copy the data if we can't */
+    if(dataFlags & (DataFlag::Owned|DataFlag::ExternallyOwned)) {
+        _in = std::move(data);
+    } else {
+        _in = Containers::Array<char>{NoInit, data.size()};
+        Utility::copy(data, _in);
+    }
+}
+
+UnsignedInt WebPImporter::doImage2DCount() const { return 1; }
+
+Containers::Optional<ImageData2D> WebPImporter::doImage2D(UnsignedInt, UnsignedInt) {
+
+    /* Create the structure that contains the WebP image data. */
+    WebPData image;
+    image.bytes = reinterpret_cast<uint8_t*>(_in.data());
+    image.size = _in.size();
+
+    /* Verify the file is a WebP image file */
+    bool isWebPImage = WebPGetInfo(image.bytes, image.size, nullptr, nullptr);
+    if(!isWebPImage) {
+        Error() << "Trade::WebPImporter::image2D(): not a webp image";
+        return {};
+    }
+
+    /* Structures for bitstream features and decoder configuration */
+    WebPDecoderConfig config;
+    WebPBitstreamFeatures bitstream = config.input;
+    CORRADE_INTERNAL_ASSERT_OUTPUT(WebPInitDecoderConfig(&config));
+
+    /* Reading the file information into config.input */
+    if(WebPGetFeatures(image.bytes, image.size, &bitstream) != VP8_STATUS_OK) {
+        Error{} << "The features couldn't be retrieved.";
+        return {};
+    }
+
+    /* Filtering animated webp files, they are subject to a different decoding process defined in demux library */
+    if(bitstream.format == 0) {
+        Error{} << "Animated files aren't supported.";
+        return {};
+    }
+
+    /* Image size */
+    const Vector2i size(bitstream.width, bitstream.height);
+
+    /* Channel number and pixel format (always 8-bit per channel) determined by alpha transparency,
+       independent of the image quality (lossy or lossless) */
+    int channels = 3;
+    PixelFormat pixelFormat = PixelFormat::RGB8Unorm;
+    WEBP_CSP_MODE colourDepth = MODE_RGB;
+
+    if(bitstream.has_alpha) {
+        channels = 4;
+        pixelFormat = PixelFormat::RGBA8Unorm;
+        colourDepth = MODE_RGBA;
+    }
+
+    /* Structure and configuration for decoding */
+    WebPDecBuffer* const outputBuffer = &config.output;
+    const std::size_t stride = 4*((size.x()*channels + 3)/4);
+    outputBuffer->u.RGBA.size = stride * size.y();
```

Comment:
```suggestion
    outputBuffer->u.RGBA.size = stride*size.y();
```

Minor code-style issue